src/ripple/overlay/impl/ProtocolMessage.h

```diff
@@ -106,80 +214,102 @@ std::pair <std::size_t, boost::system::error_code>
 invokeProtocolMessage (Buffers const& buffers, Handler& handler)
 {
     std::pair<std::size_t,boost::system::error_code> result = { 0, {} };
-    boost::system::error_code& ec = result.second;
 
-    auto const bs = boost::asio::buffer_size(buffers);
+    auto const size = boost::asio::buffer_size(buffers);
+
+    if (size == 0)
+        return result;
+
+    auto header = detail::parseMessageHeader(buffers, size);
 
-    // If we don't even have enough bytes for the header, there's no point
-    // in doing any work.
-    if (bs < Message::kHeaderBytes)
+    // If we can't parse the header then it may be that we don't have enough
+    // bytes yet, or because the message
+    if (!header)
         return result;
 
-    if (bs > Message::kMaxMessageSize)
+    /** The largest size that a message can be.
+
+        Sending a message whose size exceeds this may result in the connection
+        being dropped. A larger message size may be supported in the future or
+        negotiated as part of a protocol upgrade.
+    */
+    static std::size_t constexpr kMaxMessageSize = 64 * 1024 * 1024;
```

Comment:
Wasn't there something about support for `64MB` or `64 MiB` or `megabytes(64)` or so?

Reply:
I think so. I'll go spelunking for it.